state/upgrades.go

```diff
@@ -2246,3 +2246,72 @@ func AddControllerNodeDocs(pool *StatePool) error {
 	}
 	return nil
 }
+
+// AddSpaceIdToSpaceDocs ensures that every space document includes a
+// a sequentially generated ID.
+// It also adds a doc for the default space (ID=0).
+func AddSpaceIdToSpaceDocs(pool *StatePool) (err error) {
+	return errors.Trace(runForAllModelStates(pool, func(st *State) error {
+		// Make sure we do not add a default space to the controller model.
+		if st.isController() {
+			return nil
+		}
+
+		col, closer := st.db().GetCollection(spacesC)
+		defer closer()
+
+		type oldSpaceDoc struct {
+			SpaceId    string `bson:"spaceid"`
+			Life       Life   `bson:"life"`
+			Name       string `bson:"name"`
+			IsPublic   bool   `bson:"is-public"`
+			ProviderId string `bson:"providerid,omitempty"`
+		}
+
+		var docs []oldSpaceDoc
+		err := col.Find(nil).All(&docs)
```

Comment:
I think we might want to Sort() this somehow. I don't have a particular need to, but it does make upgrading to Ids more stable. (Something getting interrupted will create things in the same order, etc.) I think we don't have to worry about it, but something to consider.